homeassistant/components/lastfm/sensor.py

```diff
@@ -35,23 +41,46 @@ def format_track(track: Track) -> str:
     return f"{track.artist} - {track.title}"
 
 
-def setup_platform(
+async def async_setup_platform(
     hass: HomeAssistant,
     config: ConfigType,
-    add_entities: AddEntitiesCallback,
+    async_add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
-    """Set up the Last.fm sensor platform."""
-    lastfm_api = LastFMNetwork(api_key=config[CONF_API_KEY])
-    entities = []
-    for username in config[CONF_USERS]:
-        try:
-            user = lastfm_api.get_user(username)
-            entities.append(LastFmSensor(user, lastfm_api))
-        except WSError as exc:
-            LOGGER.error("Failed to load LastFM user `%s`: %r", username, exc)
-            return
-    add_entities(entities, True)
+    """Set up the Last.fm sensor platform from yaml."""
+
+    async_create_issue(
+        hass,
+        DOMAIN,
+        "deprecated_yaml",
+        breaks_in_ha_version="2023.8.0",
+        is_fixable=False,
+        severity=IssueSeverity.WARNING,
+        translation_key="deprecated_yaml",
+    )
+
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=config
+        )
+    )
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Initialize the entries."""
+
+    lastfm_api = LastFMNetwork(api_key=entry.data[CONF_API_KEY])
```

Comment:
```suggestion
    lastfm_api = LastFMNetwork(api_key=entry.options[CONF_API_KEY])
```

Reply:
Take care where data has been stored. You also need to modify the tests accordingly.